src/steps/podman.rs

```diff
@@ -0,0 +1,64 @@
+use anyhow::Result;
+
+use crate::terminal::print_separator;
+use crate::{execution_context::ExecutionContext, utils::require};
+use json;
+use log::debug;
+use std::path::Path;
+use std::process::Command;
+
+/// Returns a Vector of all containers, as a triple (repository, container name, version)
+fn list_containers(podman: &Path) -> Result<Vec<(String, String, String)>> {
+    let output = Command::new(podman).args(&["images", "--format", "json"]).output()?;
+    let output_str = String::from_utf8(output.stdout)?;
+    let json_str = json::parse(&output_str)?;
+
+    let mut retval = vec![];
+    for lentry in json_str.members() {
+        // FIXME: Bad unwraps!
+        let cont_name = lentry["Names"][0].as_str().unwrap();
+        let (remainder, tag) = cont_name.split_once(":").unwrap();
+        if let Some((registry, name)) = remainder.split_once("/") {
+            if registry == "localhost" {
+                // Don't know how to update self-built containers
+                continue;
+            }
+
+            retval.push((registry.into(), name.into(), tag.into()));
+        } else {
+            panic!("Oh nooooo!");
+        }
+    }
+
+    Ok(retval)
+}
+
+pub fn run_podman(ctx: &ExecutionContext) -> Result<()> {
+    let podman = require("podman")?;
+
+    print_separator("Podman");
+    let containers = list_containers(&podman)?;
+    debug!("Containers to inspect: {:?}", containers);
+
+    for (registry, name, tag) in containers.iter() {
+        // FIXME: Should I return the whole string from podmans output directly instead?
+        let mut container = registry.to_owned();
+        container.push('/');
+        container.push_str(name);
+        container.push(':');
+        container.push_str(tag);
+        let args = vec!["pull", &container[..]];
+
+        // TODO: How would I check whether the execution was successful?
```

Comment:
`check_run()` does it. However, it will cause the function to quit and skip the remaining contains if one pull fails. I suggest that you keep a boolean variable indicating whether any container failed and try to iterate all of them even if some fail.

Reply:
Done! I now make it return a `StepFailed` at the end if any of the partial steps failed. It now also prints debug outputs if something goes wrong. Should I maybe use `warning!` or `error!` for things that cause step failures? Do you have any rules for this?